bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Link.java

```diff
@@ -105,8 +106,19 @@ public class Link extends Control {
  * @see Widget#checkSubclass
  * @see Widget#getStyle
  */
-public Link (Composite parent, int style) {
-	super (parent, style);
+public Link(Composite parent, int style) {
+	super(parent, style);
+	/*
+	 * Accessible tool like JAWS by default only reads the hypertext link text and
+	 * not the non-link text. To make JAWS read the full text we need to tweak the
+	 * default behavior and explicitly return the full link text as below.
+	 */
+	this.getAccessible().addAccessibleListener(new AccessibleAdapter() {
+		@Override
+		public void getName(AccessibleEvent e) {
+			e.result = Link.this.getText();
```

Comment:
We can use text here directly instead of Link.this.getText().

Reply:
Thanks Lakshmi, will do the refactoring after RC1 is promoted

Reply:
#353 takes care of this.